app/javascript/components/widgets/numbers/rapierPixi.js

```diff
@@ -0,0 +1,229 @@
+import githubCommitIcon from '@images/commit.png';
+import slackMessageIcon from '@images/slack.png';
+
+export default function run_simulation(app, eventData) {
+  var githubPullCount = eventData.events.count.githubPull;
+  var githubCommitCount = eventData.events.count.githubCommit;
+  var slackMessageCount = eventData.events.count.slackMessage;
+
+  const windowSizeX = 1100;
+  const windowSizeY = 600;
+  const border = 100;
+
+  //Rapier world settings
+  const scaleFactor = 25;
+  let gravity = new RAPIER.Vector2(0.0, -9.81 * scaleFactor);
+  let world = new RAPIER.World(gravity);
+
+  const sprites = [];
+
+  //Rapier ground block (static)
+  let groundBodyDesc = RAPIER.RigidBodyDesc.fixed().setTranslation(
+    0,
+    -windowSizeY + border,
+  );
+  let groundBody = world.createRigidBody(groundBodyDesc);
+  let groundColliderDesc = RAPIER.ColliderDesc.cuboid(windowSizeX, 10);
+  world.createCollider(groundColliderDesc, groundBody);
+
+  //Rapier left wall block (static)
+  let leftWallBodyDesc = RAPIER.RigidBodyDesc.fixed().setTranslation(
+    -border,
+    0,
+  );
+  let leftWallBody = world.createRigidBody(leftWallBodyDesc);
+  let leftWallColliderDesc = RAPIER.ColliderDesc.cuboid(10, windowSizeY);
+  world.createCollider(leftWallColliderDesc, leftWallBody);
+
+  //Rapier right wall block (static)
+  let rightWallBodyDesc = RAPIER.RigidBodyDesc.fixed().setTranslation(
+    windowSizeX - border,
+    0,
+  );
+  let rightWallBody = world.createRigidBody(rightWallBodyDesc);
+  let rightWallColliderDesc = RAPIER.ColliderDesc.cuboid(10, windowSizeY);
+  world.createCollider(rightWallColliderDesc, rightWallBody);
+
+  let num = 1;
+  let numy = githubPullCount + githubCommitCount + slackMessageCount;
+  let numBodies = num * numy;
+  let rad = 8.0;
+
+  let shift = rad * 1.5 + rad;
+  let centerx = shift * (num / 2) - 400;
+  let centery = shift / 2.0;
+
+  let i, j;
+
+  //Rapier falling cubes (dynamic)
+  for (j = 0; j < numy; ++j) {
+    for (i = 0; i < num; ++i) {
+      let x = i * shift - centerx + (Math.random() * (200 - -200) + -200);
+      let y = j * shift + centery + 3.0;
+      let bodyDesc = RAPIER.RigidBodyDesc.dynamic().setTranslation(x, y);
+      let body = world.createRigidBody(bodyDesc);
+      let colliderDesc = RAPIER.ColliderDesc.ball(rad);
+      world.createCollider(colliderDesc, body);
+
+      let img = githubPullIcon;
+      var spotFound = false;
+
+      while (!spotFound) {
+        let rand = Math.floor(Math.random() * 3);
+        if (rand == 0 && githubPullCount > 0) {
+          spotFound = true;
+          githubPullCount--;
+          img = githubPullIcon;
+        } else if (rand == 1 && githubCommitCount > 0) {
+          spotFound = true;
+          githubCommitCount--;
+          img = githubCommitIcon;
+        } else if (rand == 2 && slackMessageCount > 0) {
+          spotFound = true;
+          slackMessageCount--;
+          img = slackMessageIcon;
+        }
+      }
```

Comment:
It took me a little bit to understand what this loop was doing. Maybe extract a method and prevent a potential infinite loop?